src/annotated_lcf.fun

```diff
@@ -0,0 +1,40 @@
+functor AnnotatedLcf
```

Comment:
Yes, I think that that is what you encountered before...

I personally prefer NJ's behavior (as far as ignoring types that are already uniquely determined), but I believe the Definition does not allow this. Bob and Andreas believe this to be a mistake, so far as I can tell...

Structure sharing itself is kind of bizarre, because it really only induces sharing on the types, and not on the values. In SML '90, each structure had an intensional identity, and structure sharing meant a constraint on that identity; this was removed in SML '97, and structure sharing degenerates to constraints on the type components of each structure.